--- src/VBox/Additions/linux/drm/vboxvideo_drm.c.orig	2015-05-21 17:31:03.536552946 +0300
+++ src/VBox/Additions/linux/drm/vboxvideo_drm.c	2015-05-21 17:34:06.194549985 +0300
@@ -122,7 +122,12 @@
         .open = drm_open,
         .release = drm_release,
         .unlocked_ioctl = drm_ioctl,
+# if LINUX_VERSION_CODE < KERNEL_VERSION(3, 18, 0)
+        /* This shouldn't be necessary even for old kernels as there is
+         * nothing sensible to mmap. But we play safe and keep it for
+         * legacy reasons. */
         .mmap = drm_mmap,
+# endif
         .poll = drm_poll,
 };
 #endif
